barretenberg/cpp/src/barretenberg/sumcheck/sumcheck.test.cpp

```diff
@@ -248,9 +320,43 @@ TEST_F(SumcheckTests, ProverAndVerifierSimple)
 
         auto verified = verifier_output.verified.value();
 
-        EXPECT_EQ(verified, expect_verified);
+        EXPECT_EQ(verified, false);
     };
+};
 
-    run_test(/* expect_verified=*/true);
-    run_test(/* expect_verified=*/false);
+// Define the FlavorTypes
+using FlavorTypes = testing::Types<UltraFlavor, UltraFlavorWithZK>;
+
+TYPED_TEST_SUITE(SumcheckTests, FlavorTypes);
+// Test the verifier
+TYPED_TEST(SumcheckTests, Prover)
+{
+    this->test_prover();
+}
+// Tests the prover-verifier flow
+TYPED_TEST(SumcheckTests, ProverAndVerifierSimple)
+{
+    this->test_prover_verifier_flow();
 }
+// This tests is fed an invalid circuit and checks that the verifier would output false.
+TYPED_TEST(SumcheckTests, ProverAndVerifierSimpleFailure)
+{
+    this->test_failure_prover_verifier_flow();
+}
+
+// UltraFlavor-specific tests
+using UltraFlavorSumcheckTests = SumcheckTests<UltraFlavor>;
+// Checks that polynomials are updated correctly
+TEST_F(UltraFlavorSumcheckTests, PolynomialNormalization)
+{
+    test_polynomial_normalization();
+}
+// UltraFlavorWithZK-specific tests
+using UltraFlavorWithZKSumcheckTests = SumcheckTests<UltraFlavorWithZK>;
+
+// TEST_F(UltraFlavorSumcheckTests, LibraPolynomialsNormalization)
```

Comment:
why is this commented out? Also you could consider GTEST_SKIP and then have all the text be TYPED_TEST and some skipped for a specific flavor accordingly

Reply:
don't remember what I tried to do there, thanks for pointing out. GTEST_SKIP done!